pkg/assets/retrieve.go

```diff
@@ -0,0 +1,148 @@
+	// Retrieve the asset information for the list of contract addresses
+	assetInfoList, retrievalErr := h.assetsStore.RetrieveAssetsInfo(params.AssetAddresses)
+
+	if retrievalErr != nil {
+		// If some asset information was retrieved successfully
+		if len(assetInfoList) > 0 {
+			// Create the response containing the partial asset information list
+			response := operations.RetrieveAssetInfoListPartialContentBody{
+				Assets: assetInfoList,
+			}
+
+			// Return the response with the partial list using 206 Partial Content
+			return operations.NewRetrieveAssetInfoListPartialContent().WithPayload(&response)
+		}
+
+		// If there is no retrieved asset information, return the internal server error
+		return operations.NewRetrieveAssetInfoListInternalServerError().WithPayload(
+			&models.Error{
+				Code:    "internal_server_error",
+				Message: "Failed to retrieve asset information",
+			},
+		)
+	}
+
+	// All asset information retrieved successfully
+
+	// Create the response containing the asset information list
+	response := operations.RetrieveAssetInfoListOKBody{
+		Assets: assetInfoList,
+	}
+
+	// Return the response with the full list of retrieved asset information using 200 OK
+	return operations.NewRetrieveAssetInfoListOK().WithPayload(&response)
+}
+
+
+// RetrieveAssetsInfo retrieves asset information for a given list of contract addresses concurrently.
+func (s *AssetsStore) RetrieveAssetsInfo(contractAddresses []string) ([]models.AssetInfo, error) {
+	var wg sync.WaitGroup
+	assetInfoCh := make(chan *models.AssetInfo, len(contractAddresses))
+	errCh := make(chan error, len(contractAddresses))
+
+	for _, address := range contractAddresses {
+		wg.Add(1)
+		go func(address string) {
+			defer wg.Done()
+
+			assetInfo, err := s.RetrieveAssetInfo(address)
+			if err != nil {
+				errCh <- errors.Wrapf(err, "failed to retrieve asset information for contract address %s", address)
+				return
+			}
+
+			assetInfoCh <- assetInfo
+		}(address)
+	}
+
+	wg.Wait()
+	close(assetInfoCh)
+	close(errCh)
+
+	// Collect asset information and errors
+	var assetInfos []models.AssetInfo
+	var allErrors []error
+
+	for assetInfo := range assetInfoCh {
+		assetInfos = append(assetInfos, *assetInfo)
+	}
+
+	for err := range errCh {
+		allErrors = append(allErrors, err)
+	}
+
+	// If there were errors, return them along with the retrieved asset information
+	if len(allErrors) > 0 {
+		return assetInfos, fmt.Errorf("failed to retrieve some assets: %w", gatherErrors(allErrors))
+	}
+
+	return assetInfos, nil
+}
+
+// RetrieveAssetInfo retrieves the asset information for a given contract address.
+func (s *AssetsStore) RetrieveAssetInfo(contractAddress string) (*models.AssetInfo, error) {
+	// Check if the asset information exists in the JSON
+	assetInfo, err := s.getAssetInfoFromJSON(contractAddress)
+	if err == nil {
+		// Asset information found in the JSON
+		return assetInfo, nil
+	} else if errors.Is(err, ErrAssetInfoNotFound) {
```

Comment:
how this can happen ??? if you have a token Address in the json, you should have the info associated.
If its not possible to happen, please delete the unused code to keep it simple

Reply:
if ErrAssetInfoNotFound happen, its mean asset is not in JSON, so why you want get infos about this ?  this should not happen